client/rpc-spec-v2/src/chain_head/chain_head_follow.rs

```diff
@@ -0,0 +1,591 @@
+		if !self.runtime_updates {
+			return None
+		}
+
+		let block_rt = match self.client.runtime_version_at(block) {
+			Ok(rt) => rt,
+			Err(err) => return Some(err.into()),
+		};
+
+		let parent = match parent {
+			Some(parent) => parent,
+			// Nothing to compare against, always report.
+			None => return Some(RuntimeEvent::Valid(RuntimeVersionEvent { spec: block_rt })),
+		};
+
+		let parent_rt = match self.client.runtime_version_at(parent) {
+			Ok(rt) => rt,
+			Err(err) => return Some(err.into()),
+		};
+
+		// Report the runtime version change.
+		if block_rt != parent_rt {
+			Some(RuntimeEvent::Valid(RuntimeVersionEvent { spec: block_rt }))
+		} else {
+			None
+		}
+	}
+
+	/// Get the in-memory blocks of the client, starting from the provided finalized hash.
+	fn get_init_blocks_with_forks(
+		&self,
+		info: &Info<Block>,
+	) -> Result<InitialBlocks<Block>, SubscriptionManagementError> {
+		let blockchain = self.backend.blockchain();
+		let leaves = blockchain.leaves()?;
+		let finalized = info.finalized_hash;
+		let mut pruned_forks = HashSet::new();
+		let mut finalized_block_descendants = Vec::new();
+		for leaf in leaves {
+			let tree_route = sp_blockchain::tree_route(blockchain, finalized, leaf)?;
+
+			let blocks = tree_route.enacted().iter().map(|block| block.hash);
+			if !tree_route.retracted().is_empty() {
+				pruned_forks.extend(blocks);
+			} else {
+				// Ensure a `NewBlock` event is generated for all children of the
+				// finalized block. Describe the tree route as (child_node, parent_node)
+				// Note: the order of elements matters here.
+				let parents = std::iter::once(finalized).chain(blocks.clone());
+				finalized_block_descendants.extend(blocks.zip(parents));
+			}
+		}
+
+		// Keep unique blocks.
+		let finalized_block_descendants: Vec<_> =
+			finalized_block_descendants.into_iter().unique().collect();
+
+		Ok(InitialBlocks { finalized_block_descendants, pruned_forks })
+	}
+
+	/// Generate the initial events reported by the RPC `follow` method.
+	///
+	/// Returns the initial events that should be reported directly, together with pruned
+	/// block hashes that should be ignored by the `Finalized` event.
+	fn generate_init_events(
+		&mut self,
+		info: &Info<Block>,
+	) -> Result<(Vec<FollowEvent<Block::Hash>>, HashSet<Block::Hash>), SubscriptionManagementError>
+	{
+		let init = self.get_init_blocks_with_forks(info)?;
+
+		let initial_blocks = init.finalized_block_descendants;
+
+		// The initialized event is the first one sent.
+		let finalized_block_hash = info.finalized_hash;
+		self.sub_handle.pin_block(finalized_block_hash)?;
+
+		let finalized_block_runtime = self.generate_runtime_event(finalized_block_hash, None);
+
+		let initialized_event = FollowEvent::Initialized(Initialized {
+			finalized_block_hash,
+			finalized_block_runtime,
+			runtime_updates: self.runtime_updates,
+		});
+
+		let mut in_memory_blocks = Vec::with_capacity(initial_blocks.len() + 1);
```

Comment:
Just curious, why `in_memory_blocks`?
I would change this `in_memory_blocks` name to something else.
Maybe you can change it to `finalized_block_descendants` as you've done for the struct field?